compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -141,6 +141,9 @@ class TypeComparer(initctx: Context) extends ConstraintHandling[AbsentContext] w
    */
   private [this] var leftRoot: Type = _
 
+  /** Are we forbidden from recording GADT constraints? */
+  private[this] var frozenGadt = false
```

Comment:
I wondered, can this somehow be combined with the GADTflexible mode? I guess this would mean that instead of setting the GADTflexible mode bit we create a context with a new type comparer where frozenGadt is true.

...

After looking further I see that frozenGadt and GADTflexible are not equivalent. `either` depends on GADTflexible but not frozenGadt. I assume that's intentional.

Reply:
I can see how `GADTFlexible` and `frozenGadt` can appear to have the same purpose, though I don't really think of them this way. For me, `GADTFlexible` mode fundamentally alters `isSubType(tp, pt)` to mean "what type constraints necessarily follow from `tp <: pt`?", whereas `frozenGadt` is simply a temporary flag that `TypeComparer` sets for itself, much like `ApproxState` is a set of temporary flags that help with recording constraints.

At one point, `frozenGadt` was a part of `ApproxState`. Would it help if I called it `gadtApprox` and added more information to the comment?